DIPs/DIP1xxx-JS.md

```diff
@@ -0,0 +1,109 @@
+# The Deprecation Process
+
+| Field | Value |
+|-----------------|-----------------------------------------------------------------|
+| DIP:            | xxxx |
+| Review Count:   | 0 |
+| Author:         | Jack Stouffer |
+| Implementation: | N/A |
+| Status:         | Draft |
+
+## Abstract
+
+In order to incrementally improve D or it's standard library, it's often necessary to
+mark features or functions for future removal. This document proposes a standardized
+process for language maintainers to remove public features. This process would be 
+used across DMD, Druntime, and Phobos.
+
+## Rationale
+
+There is general disagreement on the best and/or accepted way to remove public
+features. Each deprecation ends being handled slightly differently depending on
+who's handling the pull requests. Standardizing the process makes sure that
+deprecations are done very publicly and carefully, as to minimize breakage and
+to provide clear fixes for user code.
+
+## Description
+
+The following lays out the rules to follow to deprecate a part of D.
+
+A symbol or feature must not be marked for removal on a specific date, but rather on a
+specific release. This allows users to easily know if upgrading will break their
+code or not.
+
+Users must be given at least eight non-patch releases before the deprecated features
```

Comment:
ten deprecated with five undocumented would be better, since if we're actually on target for all releases, eight would only be 16 months rather than 24 like we do now. Ten would still only be 20 months if all releases are on time, but they get delayed frequently enough that that will probably push it into the 2 year range which is where we want to be. So, with 10 releases, we might be removing stuff a bit early, but with eight, it's pretty much a guarantee that we will unless releases start falling behind significantly.